lib/manageiq/active_record_connector.rb

```diff
@@ -0,0 +1,60 @@
+require "active_record"
+unless defined? DatabaseConfigurationPatch
+  require_relative "../patches/database_configuration_patch.rb"
+end
+
+module ManageIQ
+  class ActiveRecordConnector
+    # Helper class for getting the database config without needing Rails
+    #
+    # Makes use of the database_configuration_patch for the most part for
+    # grabbing this from the yaml file in config/database.yml
+    class ConnectionConfig
+      # Set a default method for handling super call in
+      # lib/patches/database_configuration_patch.rb
+      def self.database_configuration
+        if ENV["DATABASE_URL"]
+          {}
+        end
+      end
+
+      class << self
+        prepend DatabaseConfigurationPatch
+      end
+
+      def self.[](env)
+        database_configuration[env]
+      end
+    end
+
+    # Returns true if an ActiveRecord::Base.connection_config is present,
+    # otherwise returns false
+    def self.connection_exists?
+      begin
+        true if ActiveRecord::Base.connection_config
```

Comment:
Probably.  Based this code basically off of `fix_auth`, which I guess maybe wasn't the best idea since @kbrock seems to absolutely hate that code (he wrote it by the way).